packages/xdl/src/Web.js

```diff
@@ -79,39 +121,158 @@ export async function onlySupportsWebAsync(projectRoot) {
+}
+
+async function attemptToGetPackageVersionAtPathAsync(modulesPath, moduleName) {
+  const possibleModulePath = path.resolve(modulesPath, moduleName, 'package.json');
+  if (fs.existsSync(possibleModulePath)) {
+    const packageJson = await JsonFile.readAsync(possibleModulePath);
+    return packageJson.version;
+  }
+}
+
+async function getLibraryVersionAsync(projectRoot, packageName) {
+  const modulesPath = getModulesPath(projectRoot);
+  const possiblePackageVersion = await attemptToGetPackageVersionAtPathAsync(
+    modulesPath,
+    packageName
+  );
+  return possiblePackageVersion;
+}
+
+export async function ensureWebSupportAsync(projectRoot, isInteractive = true) {
+  let errors = [];
+
   const hasWebSupport = await hasWebSupportAsync(projectRoot);
   if (!hasWebSupport) {
-    throw new XDLError(ErrorCode.WEB_NOT_CONFIGURED, getWebSetupLogs());
+    if (isInteractive && (await promptToAddWebPlatform())) {
+      await addWebPlatformToAppConfig(projectRoot);
+    } else {
+      errors.push(getWebSetupLogs());
+      isInteractive = false;
+    }
   }
+
+  const results = await getMissingReactNativeWebPackagesMessageAsync(projectRoot);
+  if (results) {
+    if (
+      isInteractive &&
+      (await promptToInstallReactNativeWeb(results.dependencies, results.command))
+    ) {
+      try {
+        await installAsync(projectRoot, results.dependencies);
+        // In the case where the package.json value is different from the installed value
+        // and the installed value is valid, but the package.json version is not,
+        // then the initial install would have synchronized the values but downloaded an
+        // invalid package version. Running this twice is the only way to catch this.
+        // To test install all of the valid versions, then set expo version to 29.0.0, and delete react.
+        // Starting the flow over again will prompt to install react, then the second pass will catch expo.
+        const postResults = await getMissingReactNativeWebPackagesMessageAsync(projectRoot);
+        if (postResults) {
+          // TODO: Bacon: Maybe we should warn that there was a problem with the first pass.
+          await installAsync(projectRoot, postResults.dependencies);
+        }
+      } catch (error) {
+        throw new XDLError(
+          ErrorCode.WEB_NOT_CONFIGURED,
+          'Failed to install the required packages: ' + error.message
+        );
+      }
+    } else {
+      errors.push(results.message);
+    }
+  }
+
+  if (errors.length) {
+    errors.unshift(chalk.bold(`\nTo use Expo in the browser you'll need to:`));
+    throw new XDLError(ErrorCode.WEB_NOT_CONFIGURED, errors.join('\n'));
+  }
+}
+
+async function promptToInstallReactNativeWeb(dependencies, command) {
+  // clearConsole();
+  const question = {
+    type: 'confirm',
+    name: 'should',
+    message: `It appears you don't have all of the required packages installed. \n  Would you like to install: ${chalk.white.underline(
```

Comment:
<img width="555" alt="Screenshot 2019-04-12 at 17 58 02" src="https://user-images.githubusercontent.com/497214/56046711-89a99200-5d4c-11e9-9ac9-12bc4762c1f8.png">

Please avoid using `chalk.white` as it is invisible/illegible in terminals using a light background color.

Reply:
```suggestion
    message: `It appears you don't have all of the required packages installed. \n  Would you like to install: ${chalk.underline(
```